### PR TITLE
Allow action fields to have a class

### PR DESCRIPTION
**Pull request by bonkydog**
Hi.  Thanks for making Oat!

I added the ability to add classes to Siren action fields, as described in [version 0.6.1 of the Siren spec](https://github.com/kevinswiber/siren#class-4)

Please consider merging this into ismasan/oat .

### CODE REVIEW

File: lib/oat/adapters/siren.rb [hunk 1]
```diff
@@ -91,7 +91,11 @@ class Field
           attr_reader :data
 
           def initialize(name)
-            @data = { :name => name }
+            @data = { :name => name, :class => []}
+          end
+
+          def class(value)
```
Comment (ismasan) [inline lib/oat/adapters/siren.rb]: You're overwriting Ruby's `#class` method, which is expected to return the object's class.

Comment (bonkydog) [inline lib/oat/adapters/siren.rb]: I know, it's a little weird.  This is already being done in other parts of the Oat codebase though, for instance ```Oat::Adapters::Siren::Action#class``` .

I'd be fine if these became something like ```siren_class```.

**Comment by ismasan** [comment]
You're totally right. Annoying, as I wouldn't want to add another `#class` method but giving it a different name would be inconsistent with the one already there.

Maybe rename both something like `#_class` (I prefer that over `#siren_class`, as then we can at least have the same method name in other adapters, potentially). Then I can deprecate the old `#class` and release a new minor version.

What do you think?

**Comment by bonkydog** [comment]
I'm not crazy about leading underscores.  How about the classic ```klass```?

**Comment by ismasan** [comment]
yeah, that works too.

**Comment by ismasan** [comment]
Hi. Do we still want to change this to `#klass`?

**Comment by milgner** [comment]
This looks very interesting! Is it still active? :smile:

**Comment by DanielKuhn** [comment]
@ismasan This would be a great addition. Is it getting merged?

**Comment by ismasan** [comment]
Sorry for the delay. This is merged and released now (0.6.0).

Breaking changes: all `#class` methods in the Siren adapter have been renamed to `#klass`.

```ruby
action :close_account do |action|
  action.href "http://foo.bar.com/#{item.id}/close_account"
  action.klass 'danger'
  action.klass 'irreversible'
  action.method 'DELETE'
  action.type   'application/json'
  action.field :current_password do |field|
    field.klass 'string'
    field.type :password
    field.title 'enter password:'
  end
end
```

**Comment by ismasan** [comment]
.. And by "delay" I mean 3 years :(

**Comment by DanielKuhn** [comment]
Nevermind the delay - thanx @ismasan !
We're actually using oat for our product now, good work

**Pull request merged**